### ISSUE TITLE
[FEATURE] Make Dehacked Modified Projectiles Above Speed 32 Not Drift From Center

### ISSUE THREAD

**Issue by Ru5tK1ng** [odamex/odamex#532]
**Describe the bug**
When using DEHACKED to adjust a projectile speed, anything above 32 will cause the projectile to veer to the sides. Really noticeable with speeds 40 and above.

**Build that the bug occurred in**
Odamex 10 build.

**To Reproduce**
Set a rocket or plasmaball to a speed of 40.

**Expected behavior**
You'll see it offset as if it's locking onto monsters when there are none.

**Screenshots, NetDemos, & Crash Dumps**
https://i.imgur.com/JtblyOn.png

**Comment by rakohus** [comment]
Changing to feature request as this is actually an extension beyond vanilla behavior.

**Comment by bcahue** [comment]
I think this would require floating point coordinate calculations to fully resolve if I remember correctly.

**Comment by bcahue** [comment]
Maybe not, I think we can "borrow" Hexen's "FastProjectile" properties used in the Magic Wand and Cleric Flame, in order to achieve this.

DSDA code here:

```c++
  if (hexen)
  {
    if (th->type == HEXEN_MT_MWAND_MISSILE || th->type == HEXEN_MT_CFLAME_MISSILE)
    {                           // Ultra-fast ripper spawning missile
      th->x += (th->momx >> 3);
      th->y += (th->momy >> 3);
      th->z += (th->momz >> 3);
    }
    else
    {                           // Normal missile
      th->x += (th->momx >> 1);
      th->y += (th->momy >> 1);
      th->z += (th->momz >> 1);
    }
    if (!P_TryMove(th, th->x, th->y, false))
    {                           // Exploded immediately
      P_ExplodeMissile(th);
      return (NULL);
    }
    return (th);
  }
```
------
```c++
    // Handle movement
    if (mobj->momx || mobj->momy || (mobj->z != mobj->floorz) || mobj->momz)
    {
        xfrac = mobj->momx >> 3;
        yfrac = mobj->momy >> 3;
        zfrac = mobj->momz >> 3;
        changexy = xfrac || yfrac;
        for (i = 0; i < 8; i++)
        {
            if (changexy)
            {
                if (!P_TryMove(mobj, mobj->x + xfrac, mobj->y + yfrac, false))
                {               // Blocked move
                    P_ExplodeMissile(mobj);
                    return;
                }
            }
            mobj->z += zfrac;
            if (mobj->z <= mobj->floorz)
            {                   // Hit the floor
                mobj->z = mobj->floorz;
                P_HitFloor(mobj);
                P_ExplodeMissile(mobj);
                return;
            }
            if (mobj->z + mobj->height > mobj->ceilingz)
            {                   // Hit the ceiling
                mobj->z = mobj->ceilingz - mobj->height;
                P_ExplodeMissile(mobj);
                return;
            }
            if (changexy)
            {
                if (hexen)
                {
                    if (mobj->type == HEXEN_MT_MWAND_MISSILE && (P_Random(pr_hexen) < 128))
                    {
                        z = mobj->z - 8 * FRACUNIT;
                        if (z < mobj->floorz)
                        {
                            z = mobj->floorz;
                        }
                        P_SpawnMobj(mobj->x, mobj->y, z, HEXEN_MT_MWANDSMOKE);
                    }
                    else if (!--mobj->special1.i)
                    {
                        mobj_t *mo;
                        mobj->special1.i = 4;
                        z = mobj->z - 12 * FRACUNIT;
                        if (z < mobj->floorz)
                        {
                            z = mobj->floorz;
                        }
                        mo = P_SpawnMobj(mobj->x, mobj->y, z, HEXEN_MT_CFLAMEFLOOR);
                        if (mo)
                        {
                            mo->angle = mobj->angle;
                        }
                    }
                }
                else if (P_Random(pr_heretic) < 64)
                {
                    z = mobj->z - 8 * FRACUNIT;
                    if (z < mobj->floorz)
                    {
                        z = mobj->floorz;
                    }
                    P_SpawnMobj(mobj->x, mobj->y, z, HERETIC_MT_BLASTERSMOKE);
                }
            }
        }
    }
    // Advance the state
    if (mobj->tics != -1)
    {
        mobj->tics--;
        while (!mobj->tics)
        {
            if (!P_SetMobjState(mobj, mobj->state->nextstate))
            {                   // mobj was removed
                return;
            }
        }
    }
```

In plain English: FastProjectiles in Hexen (the Magic Wand and Cleric Flame) will step 8 times in a tic, going 1/8th of their Speed property at a time. It also bitshifts the initial speed so that the high point of a fixed number doesn't interfere with precision.

This will likely require an Odamex-specific dehacked flags property to expose this behavior to mods as its hard-coded for the weapons in Heretic/Hexen.
